cluster-state-service/internal/features/e2etasksteps/common_steps.go

```diff
@@ -179,3 +233,96 @@ func startNTasks(numTasks int, startedBy string, ecsWrapper wrappers.ECSWrapper)
 		EcsTaskList = append(EcsTaskList, ecsTask)
 	}
 }
+
+func createInstanceProfile(iamWrapper wrappers.IAMWrapper) error {
+	assumeRolePolicy := `{
+		"Version": "2012-10-17",
+		"Statement": [
+		{
+		"Effect": "Allow",
+		"Principal": {
+			"Service": "ec2.amazonaws.com"
+		},
+		"Action": "sts:AssumeRole"
+		}
+		]
+	}`
+
+	err := iamWrapper.GetRole(&roleName)
+	if err != nil {
+		err = iamWrapper.CreateRole(&roleName, &assumeRolePolicy)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = iamWrapper.GetInstanceProfile(&instanceProfileName)
+	if err != nil {
+		err = iamWrapper.CreateInstanceProfile(&instanceProfileName)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = iamWrapper.AttachRolePolicy(&policyArn, &roleName)
+	if err != nil {
+		return err
+	}
+
+	err = iamWrapper.AddRoleToInstanceProfile(&roleName, &instanceProfileName)
+
+	time.Sleep(time.Duration(5) * time.Second)
+
+	return nil
+}
+
+func deleteInstanceProfile(iamWrapper wrappers.IAMWrapper) error {
```

Comment:
Just wanted to confirm. Does this return an error if there are running ec2 instances that is using this profile?